modules/apps/frontend-data-set/frontend-data-set-test/src/testFunctional/macros/FrontendDataSet.macro

```diff
@@ -1,5 +1,58 @@
 definition {
 
+	macro chooseOptionNewFilterDropdown {
+		var key_option = ${key_option};
+
+		Type(
+			locator1 = "FrontendDataSetAdmin#NEW_FILTER_SELECT_OPTION",
+			value1 = ${key_option});
+	}
+
+	macro dateFilterSaveDates {
+		var key_dateNumber1 = ${key_dateNumber1};
+		var key_dateNumber2 = ${key_dateNumber2};
+
+		Click(
+			key_dateField = "From",
+			locator1 = "FrontendDataSetAdmin#DATEPICKER_BUTTON");
+
+		Click(
+			key_dateNumber = ${key_dateNumber1},
+			locator1 = "FrontendDataSetAdmin#CHOOSE_DATE_DATEPICKER_MENU");
+
+		Click(
+			key_dateField = "From",
+			locator1 = "FrontendDataSetAdmin#DATEPICKER_BUTTON");
+
+		Click(
+			key_dateField = "To",
+			locator1 = "FrontendDataSetAdmin#DATEPICKER_BUTTON");
+
+		Click(
+			key_dateNumber = ${key_dateNumber2},
+			locator1 = "FrontendDataSetAdmin#CHOOSE_DATE_DATEPICKER_MENU");
+
+		Click(
+			key_dateField = "To",
+			locator1 = "FrontendDataSetAdmin#DATEPICKER_BUTTON");
+
+		Click(locator1 = "FrontendDataSetAdmin#NEW_FILTER_SAVE_BUTTON");
+	}
```

Comment:
Is it possible to use YUICalendarWidget.addDate() macro instead of creating a new one?

Reply:
I guess it's not possible :( because this calendar structure is different from the YUICalendarWidget structure. for example, this dataset calendar doesn't have the yui3-calendar-content class or any other class that contains yui3 term.

Reply:
I guess it's not possible :( because this calendar structure is different from the YUICalendarWidget structure. for example, this dataset calendar doesn't have the yui3-calendar-content class or any other class that contains yui3 term.